### ISSUE TITLE
`appdash demo` sub-spans _appear_ to have duplicate/double metadata

### ISSUE THREAD

**Issue by slimsag** [sourcegraph/appdash#45]
EDIT: Updated (Mar 7) with recent findings.

Running `appdash demo` or `examples/cmd/webapp` and clicking on a sub-span from the trace, brings you to the page for that sub-span. Strangely, there _appear to be_ double results for meta-data present:

![image](https://cloud.githubusercontent.com/assets/3173176/6531439/b806a956-c3f6-11e4-9733-3eb41d18554d.png)

I tracked down why and how this occurs, and in fact it's not duplicate data at all (it's data from the client sending the request and the server responding to it) -- it's mostly working-as-intended:

1. Client sends a request to `/endpoint` with a `Span-ID` header (correct).
2. Server handling `/endpoint` checks for the `Span-ID` header and records directly to it (correct).
3. Both sets of data end up in the same span ID (also correct).

The issue is that the keys do not make a distinction between Client versus Server -- which is confusing when looking at the data. Consider trying to track down a bug in which the Go HTTP client _for some unknown reason_ always reports that the server responded with a `404` status code:

| Key | Value |
|-------|---------|
| `Response.StatusCode` | `200` |
| `Response.StatusCode` | `404` |

It's not clear to the reader whether _the server responded with 404_ or _the client received a 404_. A better presentation would be:

| Key | Value |
|-------|---------|
| `Server.Response.StatusCode` | `200` |
| `Client.Response.StatusCode` | `404` |

Aha! Now _we know_ the server responded with 200, but the client for some reason got a 404.

I've put the _Verbose Data View_ into a gist for prying eyes ([see here](https://gist.github.com/slimsag/e3d591985b052cd6c11f)).

**TL;DR**: We should probably prefix client vs. server annotations with such as clarification.

**Comment by slimsag** [comment]
I looked into this for a while yesterday, my findings were that:

- The data is not duplicate. It is the information from the HTTP server handling the request to `/endpoint` and the HTTP client sending the request to `/endpoint`.
- The data ends up in the same identical span ID.
- I believe that it will not happen if the outbound request is not the local server itself (unlikely in most real-world apps).

I'm not yet sure why the data ends up in the same span (I would expect two separate child spans with the same parent), or whether or not it is intended to end up that way.

**Comment by slimsag** [comment]
So there are two problems with trying to fix this:

1. Keys like `Server.ServerSend` look strange compared to just `Server.Send`.
2. Changing the structure itself (just to change the key names) would be API-incompatible.

Solution: There is already support for utilizing struct field tags (`trace`) with values being the key. For example ``trace:"Server.Send"`` on the `ServerSend` struct field.
